linera-core/src/client/requests_scheduler/scheduler.rs

```diff
@@ -665,6 +708,84 @@ impl<Env: Environment> RequestsScheduler<Env> {
         result
     }
 
+    /// Tries alternative peers in staggered parallel fashion.
+    ///
+    /// Launches requests to alternative peers with a stagger delay between each,
+    /// returning the first successful result. This provides a balance between
+    /// sequential (slow) and fully parallel (wasteful) approaches.
+    ///
+    /// # Arguments
+    /// - `key`: The request key (for logging)
+    /// - `peers`: List of alternative peers to try
+    /// - `operation`: The operation to execute on each peer
+    ///
+    /// # Returns
+    /// The first successful result, or the last error if all fail
+    async fn try_staggered_parallel<T, F, Fut>(
+        &self,
+        key: &RequestKey,
+        peers: Vec<RemoteNode<Env::ValidatorNode>>,
+        operation: &F,
+        staggered_delay_ms: u64,
+    ) -> Result<T, NodeError>
+    where
+        T: Send,
+        F: Fn(RemoteNode<Env::ValidatorNode>) -> Fut,
+        Fut: Future<Output = Result<T, NodeError>>,
+    {
+        use futures::stream::{FuturesUnordered, StreamExt};
+        use linera_base::time::{timer::sleep, Duration};
+
+        let mut futures = FuturesUnordered::new();
+
+        // Create futures with staggered delays
+        for (idx, peer) in peers.into_iter().enumerate() {
+            let delay = Duration::from_millis(staggered_delay_ms * idx as u64);
```

Comment:
I wonder if the duration should increase quadratically: The more peers we are already trying the more likely it is that the request is simply a very slow one, and it's not the peers' fault.